ledger/participant-integration-api/src/main/resources/db/migration/postgres-appendonly/V118__add_user_managment.sql

```diff
@@ -0,0 +1,19 @@
+--  Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+--  SPDX-License-Identifier: Apache-2.0
+
+
+
+CREATE TABLE participant_users (
+    internal_id         SERIAL          PRIMARY KEY,
+    user_id             TEXT            NOT NULL UNIQUE,
+    primary_party       TEXT,
+    created_at          BIGINT          NOT NULL
```

Comment:
```suggestion
    created_at          BIGINT          NOT NULL   DEFAULT now()
```

Reply:
Yeah, I was too quick. Here's the conversion that we used in the past: https://github.com/digital-asset/daml/blob/main/ledger/participant-integration-api/src/main/resources/db/migration/postgres-appendonly/V111__timestamp_to_bigint.sql#L26

I'd expect that to work here was well.

Reply:
Digging deeper I noticed that Postgres converts via double, which will lead to rounding errors for the microsecond part starting around year `2^53 / 1e6 / 3600 / 24 / 265 + 1970 ~= 2363`. This usecase here would be OK, or we could use the correct version
```
select extract(epoch from '9999-01-01 10:10:10.123456'::timestamp)::bigint + extract(microseconds from '9999-01-01 10:10:10.123456'::timestamp)::bigint;
```
However, I believe we are better served by following the simple principle that all time manipulation happens in DB client code. It removes one source of bugs. It does though assume that all clients have synchronized clocks when creating new timestamps. That requirement is present in other cases as well though.

Reply:
@mziolekda : I suggest we discuss what trade-off we want to take.